Automattic/woocommerce-payments
# Add dispute resolution task

Fixes #2712

#### Changes proposed in this Pull Request


This PR adds:
* a dispute resolution task for each dispute that the users have.
* a delete option to completed tasks ([related PR](https://github.com/woocommerce/woocommerce-admin/pull/7300))
* an ellipsis for each dispute resolution task ([related PR](https://github.com/Automattic/woocommerce-payments/issues/2237)).






#### Open disputes

![screenshot-localhost_8082-2021 08 19-14_53_33](https://user-images.githubusercontent.com/1314156/130120502-e756895f-e84f-4574-8838-7f97c02b96b1.png)


#### All disputes completed

![screenshot-localhost_8082-2021 08 19-14_58_26](https://user-images.githubusercontent.com/1314156/130120547-db492f28-5dbc-458f-8ffd-98104cb196ba.png)

#### Testing instructions




* Checkout this branch and go to `Payments` > `Overview`.
* Verify that the `disputes` task is not visible when there aren't any disputes.
* Create a new dispute. You can do it by buying one product with one of the [Stripe test disputes cards](https://stripe.com/docs/testing#disputes) (E.g.: `4000000000001976` use any date in the future and fill out `CVC` with any number).
* Verify that the `disputes` task is displayed correctly in the `Things to do` task list.
* Click on the card. You should be redirected to the dispute's overview screen.
* Click on the dispute you had created and accept it.
* Now the `disputes` task should be shown as completed.

-------------------


### Acceptance criteria

- [ ] When there are open disputes, a single task will be shown with the total number of open disputes
- [ ] When there are only closed disputes, a single completed task will be shown with the total number of closed disputes
- [ ] When there are no disputes at all in the system, no task will be shown
- [ ] Clicking on the task will open the disputes overview screen
- [ ] A Tracks event will be logged when the task is clicked on

**Post merge**
